tachyon/math/polynomials/univariate/lagrange_interpolation.h

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_MATH_POLYNOMIALS_UNIVARIATE_LAGRANGE_INTERPOLATION_H_
+#define TACHYON_MATH_POLYNOMIALS_UNIVARIATE_LAGRANGE_INTERPOLATION_H_
+
+#include <utility>
+#include <vector>
+
+#include "tachyon/base/containers/container_util.h"
+#include "tachyon/base/parallelize.h"
+#include "tachyon/math/polynomials/univariate/univariate_polynomial.h"
+
+namespace tachyon::math {
+
+template <size_t MaxDegree, typename Container>
+bool LagrangeInterpolate(
+    const Container& points, const Container& evals,
+    UnivariateDensePolynomial<typename Container::value_type, MaxDegree>* ret) {
+  using F = typename Container::value_type;
+  using Poly = UnivariateDensePolynomial<F, MaxDegree>;
+  using Coeffs = UnivariateDenseCoefficients<F, MaxDegree>;
+
+  if (points.size() != evals.size() || points.size() != MaxDegree + 1) {
+    return false;
+  }
+
+  if (points.size() == 1) {
+    *ret = Poly(Coeffs({evals[0]}));
+    return true;
+  }
+
+  // points = [x₀, x₁, ..., xₙ]
+  // denoms[i] = 1 / (xᵢ - x₀)(xᵢ - x₁)...(xᵢ - xₙ)
+  std::vector<F> denoms = base::CreateVector(points.size(), F::One());
+  base::Parallelize(denoms, [&points](absl::Span<F> chunk) {
+    for (size_t i = 0; i < chunk.size(); ++i) {
+      for (size_t j = 0; j < points.size(); ++j) {
+        if (j != i) {
+          chunk[i] *= (points[i] - points[j]);
+        }
+      }
+    }
+  });
+
+  F::BatchInverseInPlace(denoms);
```

Comment:
Could you create a single thread version of `F::BatchInverseInPlace()` and do together inside a loop of `base::Parallelize` above

Reply:
I think that would be equivalent to Inverse each element in a for loop:
```suggestion
base::Parallelize(denoms, [&points](absl::Span<F> chunk) {
    for (size_t i = 0; i < chunk.size(); ++i) {
      for (size_t j = 0; j < points.size(); ++j) {
        if (j != i) {
          chunk[i] *= (points[i] - points[j]);
        }
      }
      chunk[i].InverseInPlace();
    }
  });
```
Is it different from using a single thread `BatchInverseInPlace()`?

Reply:
It's different, because it computes expensive inverse for each single field. Maybe you can leave a comment!